formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Encoding.kt

```diff
@@ -278,6 +280,7 @@ internal open class CborReader(private val cbor: Cbor, protected val decoder: Cb
             @Suppress("UNCHECKED_CAST")
             decoder.nextByteString() as T
         } else {
+            decodeByteArrayAsByteString = decodeByteArrayAsByteString || deserializer.descriptor.isInlineByteString()
```

Comment:
This looks a bit suspicious. If the current descriptor is not a ByteArraySerializer().descriptor and not a value class over `@ByteString ByteArray`, why we should still try to decode it as ByteArray? 

If this is to support cases when value class property is not annotated itself, like this:
```kotlin
@Serializable data class Outer(@ByteString val i: InnerValue)

@Serializable 
@JvmInline value class InnerValue(val b: ByteArray)
```

Then I don't see tests for it.

Reply:
> I'm not sure if that is wanted
I'm also unsure because I didn't see any requests for that. At the first look, it sounds reasonable, but I do not use byte strings that often